pkg/render/istio/istio_test.go

```diff
@@ -0,0 +1,855 @@
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{},
+				},
+			}
+
+			err = component.ResolveImages(imageSet)
+			Expect(err).Should(HaveOccurred())
+		})
+	})
+
+	Describe("Objects", func() {
+		var (
+			crds      *istio.IstioComponentCRDs
+			component *istio.IstioComponent
+			err       error
+		)
+
+		BeforeEach(func() {
+			crds, component, err = istio.Istio(cfg)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			imageSet := &operatorv1.ImageSet{
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{
+						{Image: "tigera/istio-pilot", Digest: "sha256:test-pilot-digest"},
+						{Image: "tigera/istio-install-cni", Digest: "sha256:test-cni-digest"},
+						{Image: "tigera/istio-ztunnel", Digest: "sha256:test-ztunnel-digest"},
+						{Image: "tigera/istio-proxyv2", Digest: "sha256:test-proxyv2-digest"},
+					},
+				},
+			}
+			err = component.ResolveImages(imageSet)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should return CRDs", func() {
+			objsToCreate, objsToDelete := crds.Objects()
+			Expect(objsToCreate).NotTo(BeEmpty())
+			Expect(objsToDelete).To(BeEmpty())
+		})
+
+		It("should return objects to create", func() {
+			objsToCreate, objsToDelete := component.Objects()
+			Expect(objsToCreate).NotTo(BeEmpty())
+			Expect(objsToDelete).To(BeEmpty())
+		})
+
+		It("should include network policies", func() {
+			objsToCreate, _ := component.Objects()
+
+			// Verify istiod policy
+			istiodPolicy, err := rtest.GetResourceOfType[*v3.NetworkPolicy](objsToCreate, istio.IstioIstiodPolicyName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(istiodPolicy.Spec.Tier).To(Equal(networkpolicy.TigeraComponentTierName))
+			Expect(istiodPolicy.Spec.Selector).To(Equal(networkpolicy.KubernetesAppSelector(istio.IstioIstiodDeploymentName)))
+
+			// Verify istio-cni policy
+			cniPolicy, err := rtest.GetResourceOfType[*v3.NetworkPolicy](objsToCreate, istio.IstioCNIPolicyName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(cniPolicy.Spec.Tier).To(Equal(networkpolicy.TigeraComponentTierName))
+			Expect(cniPolicy.Spec.Selector).To(Equal(networkpolicy.KubernetesAppSelector(istio.IstioCNIDaemonSetName)))
+
+			// Verify ztunnel policy
+			ztunnelPolicy, err := rtest.GetResourceOfType[*v3.NetworkPolicy](objsToCreate, istio.IstioZTunnelPolicyName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(ztunnelPolicy.Spec.Tier).To(Equal(networkpolicy.TigeraComponentTierName))
+			Expect(ztunnelPolicy.Spec.Selector).To(Equal(networkpolicy.KubernetesAppSelector(istio.IstioZTunnelDaemonSetName)))
+		})
+
+		It("should include Istiod deployment", func() {
+			objsToCreate, _ := component.Objects()
+
+			deployment, err := rtest.GetResourceOfType[*appsv1.Deployment](objsToCreate, istio.IstioIstiodDeploymentName, istio.IstioNamespace)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(deployment.Name).To(Equal(istio.IstioIstiodDeploymentName))
```

Comment:
These assertions on Name / Namespace are redundant - we're using the name and namespace to query the object in `GetResourceOfType`